src/tracing/default.rs

```diff
@@ -21,39 +30,56 @@ impl Config {
             log_level,
             span_events,
             color,
+            log_format,
         } = other;
 
         self.log_level.replace_if_some(log_level);
         self.span_events.replace_if_some(span_events);
         self.color.replace_if_some(color);
+        self.log_format.replace_if_some(log_format);
     }
 }
 
-#[rustfmt::skip] // `rustfmt` formats this into unreadable single line
-pub type Logger<S> = filter::Filtered<
-    Option<fmt::Layer<S>>,
-    filter::EnvFilter,
-    S,
->;
+pub type Logger<S> =
+    filter::Filtered<Option<Box<dyn Layer<S> + Send + Sync>>, filter::EnvFilter, S>;
 
 pub fn new_logger<S>(config: &Config) -> Logger<S>
 where
     S: tracing::Subscriber + for<'span> registry::LookupSpan<'span>,
 {
-    let layer = new_layer(config);
+    let layer: Box<dyn Layer<S> + Send + Sync> = match config.log_format {
+        Some(LogFormat::Json) => Box::new(new_layer_with_json(config)),
+        _ => Box::new(new_layer(config)),
```

Comment:
I'd prefer explicit matching here:

```suggestion
        Some(LogFormat::Json) => Box::new(new_layer_with_json(config)),
        Some(LogFormat::Default) | None => Box::new(new_layer(config)),
```

Not super critical here, but probably something nice to get used to.

Reply:
agreed, its more readable.